samza-core/src/main/java/org/apache/samza/operators/impl/SendUpdateToTableOperatorImpl.java

```
@@ -0,0 +1,108 @@
+  private static final Logger LOG = LoggerFactory.getLogger(SendUpdateToTableOperatorImpl.class);
+
+  private final SendUpdateToTableOperatorSpec<K, V, U> sendUpdateToTableOpSpec;
+  private final ReadWriteTable<K, V, U> table;
+
+  public SendUpdateToTableOperatorImpl(SendUpdateToTableOperatorSpec<K, V, U>  sendUpdateToTableOpSpec, Context context) {
+    this.sendUpdateToTableOpSpec = sendUpdateToTableOpSpec;
+    this.table = context.getTaskContext().getTable(sendUpdateToTableOpSpec.getTableId());
+  }
+
+  @Override
+  protected void handleInit(Context context) {
+  }
+
+  @Override
+  protected CompletionStage<Collection<KV<K, UpdateMessage<U, V>>>> handleMessageAsync(KV<K, UpdateMessage<U, V>> message,
+      MessageCollector collector, TaskCoordinator coordinator) {
+    final CompletableFuture<Void> updateFuture = table.updateAsync(message.getKey(), message.getValue().getUpdate(),
+        sendUpdateToTableOpSpec.getArgs());
+
+    return updateFuture
+        .handle((result, ex) -> {
+          if (ex == null) {
+            return false;
+          } else if (!(ex.getCause() instanceof RecordDoesNotExistException)) {
+            throw new SamzaException("Update failed with exception: ", ex);
+          } else {
+            return message.getValue().getDefault() != null;
+          }
+        })
+        .thenCompose(shouldPutDefault -> {
+          // If update fails for a given key due to a RecordDoesNotExistException exception thrown and a default is
+          // provided, then attempt to PUT a default record for the key and then apply the update
+          if (shouldPutDefault) {
+            final CompletableFuture<Void> putFuture = table.putAsync(message.getKey(), message.getValue().getDefault(),
+                sendUpdateToTableOpSpec.getArgs());
+            return putFuture
+                .exceptionally(ex -> {
+                  LOG.error("PUT default failed due the following exception: ", ex);
```

Change this to warning since we are still going to process to do the update. Please also change the log message to say we are ignoring this error and process with updating.